src/Neo/Persistence/MemorySnapshot.cs

```diff
@@ -60,12 +60,17 @@ public void Put(byte[] key, byte[] value)
         /// <inheritdoc/>
         public IEnumerable<(byte[] Key, byte[] Value)> Seek(byte[]? keyOrPrefix, SeekDirection direction = SeekDirection.Forward)
         {
+            keyOrPrefix ??= [];
+            if (direction == SeekDirection.Backward && keyOrPrefix.Length == 0) yield break;
+
             var comparer = direction == SeekDirection.Forward ? ByteArrayComparer.Default : ByteArrayComparer.Reverse;
             IEnumerable<KeyValuePair<byte[], byte[]>> records = _immutableData;
             if (keyOrPrefix?.Length > 0)
```

Comment:
```suggestion
            if (keyOrPrefix.Length > 0)
```
`keyOrPrefix` is not null because of line 63